nrf52840-hal/src/ieee802154.rs

```diff
@@ -0,0 +1,630 @@
+    _25 = 75,
+    /// 2_480 MHz
+    _26 = 80,
+}
+
+/// Transmission power in dBm (decibel milliwatt)
+// TXPOWERA enum minus the deprecated Neg30dBm variant and with better docs
+#[derive(Clone, Copy, PartialEq)]
+pub enum TxPower {
+    /// +8 dBm
+    Pos8dBm,
+    /// +7 dBm
+    Pos7dBm,
+    /// +6 dBm (~4 mW)
+    Pos6dBm,
+    /// +5 dBm
+    Pos5dBm,
+    /// +4 dBm
+    Pos4dBm,
+    /// +3 dBm (~2 mW)
+    Pos3dBm,
+    /// +2 dBm
+    Pos2dBm,
+    /// 0 dBm (1 mW)
+    _0dBm,
+    /// -4 dBm
+    Neg4dBm,
+    /// -8 dBm
+    Neg8dBm,
+    /// -12 dBm
+    Neg12dBm,
+    /// -16 dBm
+    Neg16dBm,
+    /// -20 dBm (10 μW)
+    Neg20dBm,
+    /// -40 dBm (0.1 μW)
+    Neg40dBm,
+}
+
+impl TxPower {
+    fn _into(self) -> TXPOWER_A {
+        match self {
+            TxPower::Neg40dBm => TXPOWER_A::NEG40DBM,
+            TxPower::Neg20dBm => TXPOWER_A::NEG20DBM,
+            TxPower::Neg16dBm => TXPOWER_A::NEG16DBM,
+            TxPower::Neg12dBm => TXPOWER_A::NEG12DBM,
+            TxPower::Neg8dBm => TXPOWER_A::NEG8DBM,
+            TxPower::Neg4dBm => TXPOWER_A::NEG4DBM,
+            TxPower::_0dBm => TXPOWER_A::_0DBM,
+            TxPower::Pos2dBm => TXPOWER_A::POS2DBM,
+            TxPower::Pos3dBm => TXPOWER_A::POS3DBM,
+            TxPower::Pos4dBm => TXPOWER_A::POS4DBM,
+            TxPower::Pos5dBm => TXPOWER_A::POS5DBM,
+            TxPower::Pos6dBm => TXPOWER_A::POS6DBM,
+            TxPower::Pos7dBm => TXPOWER_A::POS7DBM,
+            TxPower::Pos8dBm => TXPOWER_A::POS8DBM,
+        }
+    }
+}
+
+impl<'c> Radio<'c> {
+    /// Initializes the radio for IEEE 802.15.4 operation
+    pub fn init<L, LSTAT>(radio: RADIO, _clocks: &'c Clocks<ExternalOscillator, L, LSTAT>) -> Self {
+        let mut radio = Self {
+            radio,
+            _clocks: &(),
+        };
+
+        // go to a known state
+        radio.disable();
+
+        // clear any event interesting to us
+        radio.radio.events_disabled.reset();
+        radio.radio.events_end.reset();
+
+        radio.radio.mode.write(|w| w.mode().ieee802154_250kbit());
+
+        // NOTE(unsafe) radio is currently disabled
+        unsafe {
+            radio.radio.pcnf0.write(|w| {
+                w.s1incl()
+                    .clear_bit() // S1 not included in RAM
+                    .plen()
+                    ._32bit_zero() // 32-bit zero preamble
+                    .crcinc()
+                    .include() // LENGTH field includes CRC (2 bytes)
+                    .cilen()
+                    .bits(0) // no code indicator
+                    .lflen()
+                    .bits(8) // length = 8 bits (but highest bit is reserved and must be `0`)
+                    .s0len()
+                    .clear_bit() // no S0
+                    .s1len()
+                    .bits(0) // no S1
+            });
+
+            radio.radio.pcnf1.write(|w| {
+                w.maxlen()
+                    .bits(Packet::MAX_LEN + 2 /* CRC */) // payload length
```

Comment:
This line is a little confusing. When we configure hardware, we usually work on PHY layer. Max PSDU length is 127 and it includes CRC. I would change this constant to `MAX_PSDU_LEN = 127` and use it here without `+ 2`. It would be unambiguous.